src/formElementPlugins/listBuilder/src/displayComponents/ListTableLayout.tsx

```diff
@@ -1,35 +1,61 @@
 import React from 'react'
 import { Table, TableHeaderCell, TableCell } from 'semantic-ui-react'
 import { ListLayoutProps } from '../types'
+import { TableCellMobileLabelWrapper } from '../../../../utils/tables/TableCellMobileLabelWrapper'
 import { TemplateElement } from '../../../../utils/generated/graphql'
+import { useViewport } from '../../../../contexts/ViewportState'
 
 const ListTableLayout: React.FC<ListLayoutProps & { excludeColumns: string[] }> = ({
   listItems,
   inputFields,
   editItem = () => {},
   isEditable = true,
   excludeColumns,
+  hideFromMobileIfEmpty,
+  minMobileLabelWidth,
+  maxMobileLabelWidth,
 }) => {
+  const { isMobile } = useViewport()
   const displayFields = (inputFields as TemplateElement[]).filter(
     ({ code, title }) => !(excludeColumns.includes(code) || excludeColumns.includes(title ?? ''))
   )
+
+  const hideIfEmptyFields = (() => {
+    if (typeof hideFromMobileIfEmpty === 'boolean' || hideFromMobileIfEmpty === undefined)
+      return hideFromMobileIfEmpty ? displayFields.map((el) => el.code) : []
+
+    return hideFromMobileIfEmpty
+  })()
+
   return (
-    <Table celled selectable={isEditable}>
-      <Table.Header>
-        <Table.Row>
-          {displayFields.map(({ title }) => (
-            <TableHeaderCell key={`list-header-field-${title}`}>{title}</TableHeaderCell>
-          ))}
-        </Table.Row>
-      </Table.Header>
+    <Table celled={!isMobile} stackable selectable={isEditable}>
```

Comment:
Same basic idea as what I did on the other tables -- hide the header, wrap the cells in a "Labeller", and handle optional "hidden" fields.